pkg/controllers/plugin/plugin_workload_status_controller.go

```diff
@@ -0,0 +1,325 @@
+//+kubebuilder:rbac:groups=greenhouse.sap,resources=clusters;teams,verbs=get;list;watch
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *WorkLoadStatusReconciler) SetupWithManager(name string, mgr ctrl.Manager) error {
+	r.Client = mgr.GetClient()
+	r.kubeClientOpts = []clientutil.KubeClientOption{
+		clientutil.WithRuntimeOptions(r.KubeRuntimeOpts),
+		clientutil.WithPersistentConfig(),
+	}
+	r.recorder = mgr.GetEventRecorderFor(name)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&greenhousev1alpha1.Plugin{}).
+		Complete(r)
+}
+
+func (r *WorkLoadStatusReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	var plugin = new(greenhousev1alpha1.Plugin)
+	var releaseStatus = new(ReleaseStatus)
+	if err := r.Client.Get(ctx, req.NamespacedName, plugin); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if plugin.DeletionTimestamp != nil && controllerutil.ContainsFinalizer(plugin, greenhouseapis.FinalizerCleanupHelmRelease) {
+		return ctrl.Result{}, nil
+	}
+
+	pluginStatus := initPluginStatus(plugin)
+	clusterAccessReadyCondition, restClientGetter := r.initClientGetter(ctx, *plugin, plugin.Status)
+	pluginStatus.StatusConditions.SetConditions(clusterAccessReadyCondition)
+	if !clusterAccessReadyCondition.IsTrue() {
+		return ctrl.Result{RequeueAfter: 10 * time.Minute}, fmt.Errorf("cannot access cluster: %s", clusterAccessReadyCondition.Message)
+	}
+
+	objClient, err := clientutil.NewK8sClientFromRestClientGetter(restClientGetter)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	helmRelease, err := helm.GetReleaseForHelmChartFromPlugin(ctx, restClientGetter, plugin)
+	// Skipping plugins that don't have a helm release, it could happen if the plugin is UI only
+	if err == nil && plugin.Status.GetConditionByType(greenhousev1alpha1.HelmReconcileFailedCondition).Status == metav1.ConditionFalse {
+		releaseStatus.ReleaseName = plugin.Name
+		releaseStatus.ReleaseNamespace = plugin.Spec.ReleaseNamespace
+		switch {
+		// If the release namespace is not set, use the plugin namespace
+		case plugin.Spec.ReleaseNamespace == "":
+			releaseStatus.ReleaseNamespace = plugin.Namespace
+		// If the plugin namespace is not set, use the namespace from the plugin metadata
+		case plugin.Namespace == "":
+			releaseStatus.ReleaseNamespace = plugin.ObjectMeta.Namespace
+		}
+		releaseStatus.ClusterName = plugin.Spec.ClusterName
+		releaseStatus.HelmStatus = helmRelease.Info.Status.String()
+		objectMap, _ := helm.ObjectMapFromManifest(restClientGetter, plugin.Namespace, helmRelease.Manifest, &helm.ManifestObjectFilter{})
+
+		for key := range objectMap {
+			if key.GVK.Version == "v1" &&
+				(key.GVK.Group == "apps" || key.GVK.Group == "batch") ||
+				(key.GVK.Group == monitoringAPIGroupName && key.GVK.Kind == "Alertmanager") {
+				getStatusPayload(ctx, releaseStatus, objClient, key.Name, releaseStatus.ReleaseNamespace, key.GVK)
+			}
```

Comment:
This will overwrite the releaseStatus in every iteration. That means if the last resource is ready then the status will show ready.
I think this needs to collect all the resources and then make that into a human-readable string that can be set as the message.

Maybe something like this?
```text
Following workload resources are not ready: [ "<kind>/<name>", "deployment/greenhouse-controller-manager", ...]
```

Reply:
This already poInts to the correct location to start debugging, then it is maybe not necessary to also list all the failing pods but keep it at the highest level resources: https://github.com/cloudoperators/greenhouse/pull/398/files#diff-998fb0f447e0bc4edfe14d5d70ec13c1243eba9a3570e8f9226383c5df949fbcR235